src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -148,7 +164,8 @@ public void setUp() {
 
     @Test
     public void testIsProperUtilsClass() throws ReflectiveOperationException {
-        assertUtilsClassHasPrivateConstructor(Main.class);
+        assertUtilsClassHasPrivateConstructor(Main.class,
+                TestUtils.CheckConstructorIsPrivate.FALSE);
```

Comment:
Why don't we just make 2 methods?
`assertUtilsClassHasPrivateConstructor` and `assertUtilsClassHasNonPrivateConstructor`?
I dislike enums with values `TRUE`/`FALSE`. You might as well use `Boolean`.